### PR TITLE
internal/fakecgo: clean up gen.go

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
The `go generate` command adds the `generate` build tag when searching for pragmas to run. This means the pragma doesn't need to be in a separate file.

I also moved formatting into the code itself using the `go/format` package and cleaned up the code into a template.

### CODE REVIEW

**Review by hajimehoshi — COMMENTED** [review]
I didn't think including `//go:generate` in a generating script is a conventional way.

I found these patterns in the Go repository

* Use `ignore` (the same as the current way): https://cs.opensource.google/go/go/+/master:src/image/color/palette/generate.go
* Separate a generating script in another directory like `_gen`: https://cs.opensource.google/go/go/+/master:src/cmd/compile/internal/ssa/generate.go
* Use an external command: https://cs.opensource.google/go/go/+/master:src/html/template/context.go

So, I don't think we have to change the way. Rather, I'm worried that now `internal/fakecgo` cannot be compiled with `go build -tags=generate` due to the mix of package names ([`ignore` is conventional](https://pkg.go.dev/cmd/go#hdr-Build_constraints) for such a package name by the way). Thoughts?

**Comment by TotallyGamerJet** [comment]
I don't think many people use `go:build generate` as a normal build tag outside of `go generate` but I don't know how to search all public repos for it.

I can put it back if you want. The only thing is doing it that way means only darwin and freebsd can generate it since the `doc.go` has a build tag. It has one so that it is a compilation error to import `fakecgo` on other platforms. I can add linux though.

**Comment by hajimehoshi** [comment]
Sure, let's put it back to `//go:build ignore`. Thanks,

File: internal/fakecgo/doc.go [hunk 1]
```diff
@@ -30,4 +30,3 @@
 package fakecgo
 
 //go:generate go run gen.go
-//go:generate gofmt -s -w symbols.go
```
Comment (hajimehoshi) [inline internal/fakecgo/doc.go:33]: Just to confirm: Don't we need this?

**Review by hajimehoshi — APPROVED** [review]
LGTM % comments

File: internal/fakecgo/doc.go (hunk 1 above)
Comment (TotallyGamerJet) [reply internal/fakecgo/doc.go:33]: Correct it's formatted using `go/format` in `gen.go`

**Pull request merged**